ocaml/xapi/xapi_network_attach_helpers.ml

```diff
@@ -74,14 +74,15 @@ let assert_can_attach_network_on_host ~__context ~self ~host =
 
 let assert_can_see_named_networks ~__context ~vm ~host reqd_nets =
   let is_network_available_on host net =
-    (* has the network been actualised by one or more PIFs? *)
+    (* has the network been actualised by one or more PIFs, or is managed by xapi?*)
     let pifs = Db.Network.get_PIFs ~__context ~self:net in
-    if pifs <> [] then begin
+    let managed = Db.Network.get_managed ~__context ~self:net in
+    if pifs <> [] || (not managed) then begin
```

Comment:
This is confusing, because non-managed networks never have PIFs. Better add `|| not managed` to the if-statement in L87.